### ISSUE TITLE
orphaned canvas layout on QtViewer

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#1191]
## 🐛 Bug

Was just writing up a quick response to [this image.sc question](https://forum.image.sc/t/how-can-i-get-access-to-the-image-view-for-custom-gui-elements/37203), and found this segment of code troublesome:

https://github.com/napari/napari/blob/149bd8cd88f44b0de415186f47c7bc3a015e0e01/napari/_qt/qt_viewer.py#L153-L159

basically, in the `QtViewer`, we construct a widget and a layout, but we then let it get garbage collected... so it's impossible to get access to the layout that holds the main canvas without jumping through hoops (made more difficult because that temporary widget doesn't have a parent).  Would it be ok to save that widget?

**Comment by sofroniewn** [comment]
Yes, this is fine - could make a dedicated fix PR or do it as part of #855 and @justinelarsen's work on https://github.com/napari/napari/issues/1132#issuecomment-622162235
